### DISCUSSION TITLE
Is it posibble to create dynamic dependsOn values?

### DISCUSSION THREAD

**Discussion by juangon** [operator-framework/java-operator-sdk#1732] [Q&A]
Hi!
I have a DependentResource (Ingress) that depending on other dependendant's reconcilePrecondition should use a dependsOn list or other different (because some depedent will exists or not). Is it possible to create a dynamic "dependsOn" list for some Depedent resource?

Thanks very much!

**Comment by metacosm** [comment]
I'm not sure I understand… Could you maybe detail your use case more, please? Shouldn't your dependent just implement a `reconcilePrecondition` to say that it should only be created if some other dependent exists?

**Reply by juangon** [reply]
Yes but in my case I am talking about a resource that would depend on different resources, so that will happen after reconcilePrecondition holds true

**Reply by metacosm** [reply]
Could you detail the use case more concretely, please?

**Reply by juangon** [reply]
Ok, so I have 3 Custom resources, `A`,`B`,`C`. Each `B` and `C` both have `reconcilePrecondition`, so won't exist always. And `A`resource should depend on `B`or `C` but not both as they won't exists always.

**Reply by csviri** [reply]
A way to solve this is add a ReconcilePrecodintion on A. Where you check if B or C exists.

**Reply by csviri** [reply]
It's a little hack I guess :) but I don't have better idea.

**Reply by juangon** [reply]
But `A`should exists always, it shouldn't have `reconcilePrecondition` in this case

**Reply by juangon** [reply]
The issue here is that it should depend on different lists depending on which of `B`or `C`exists.

**Reply by csviri** [reply]
Yeah, but basically the answer is the same, don't dependent with A neither on B or C, just check if any of those resources exists in a precondition.

**Reply by csviri** [reply]
This is less efficient, since it will take two reconciliation to get to a state where you could go with one otherwise (at least in general). Other options is not use workflows, just standalone dependent resources, and implement a custom logic.

**Reply by metacosm** [reply]
Yes, this should be doable with a reconcile precondition on `A` where the precondition is checking which resources it needs to wait on based on whether `B` or `C` are created. At least, that's what I would try first.

**Reply by juangon** [reply]
So the reconcilePrecondition is evaluated multiple times, not once right? I that case there I could check the status, not only the spec

**Reply by metacosm** [reply]
Hmm, you probably shouldn't store *external* state in the status… See https://javaoperatorsdk.io/docs/dependent-resources#external-state-tracking-dependent-resources (but maybe that's not what you're doing).
To be clearer, it's fine to make decisions based on resources' status (like, for example, if a pod is ready or not).

**Reply by juangon** [reply]
Oh sure, I mean I could check if dependent resource status has some expected status so the condition is true ir false

**Reply by csviri** [reply]
> So the reconcilePrecondition is evaluated multiple times, not once right? I that case there I could check the status, not only the spec

I mean it will be triggered again (the whole reconiliation) anyway if some of other resources (B,C) changes.
